### ISSUE TITLE
Maskborders are out of alignment with large radius

### ISSUE THREAD

**Issue by subliminalvoice** [domoritz/leaflet-maskcanvas#17]
Great, easy to use plugin, however if you use a large radius (50000 meters +) the edges of of the circle masks are out of alignment. 

![large_radius_issue](https://cloud.githubusercontent.com/assets/11541762/6780077/5b031cfc-d162-11e4-8398-f0838650e8d7.jpg)

Any hints how to fix this? The misalignment seems to be happening between the tiles...

Thanks for your help,
Simon

**Comment by shaula** [comment]
I can confirm this behavior (also with the plugin for Leaflet 1.0-dev). Unfortunately I also got no idea how to fix this (I'm not comfortable with projection stuff).

**Comment by casualcoder2** [comment]
I was playing with this today and can confirm the problems the others have found.

What I did was to extend this to be able to draw donut shapes by adding an extra inner radius parameter.
While it worked perfectly for smaller shapes (just a few kilometers radius),
![2](https://user-images.githubusercontent.com/47149399/51917365-5014ec00-23d7-11e9-87b0-4fd988e04c32.png)
it seems to break when shapes are so big that they span multiple tiles.
As people noted above, the edges are misaligned:
![1](https://user-images.githubusercontent.com/47149399/51917463-8488a800-23d7-11e9-83e8-e2fc80fc359e.png)
However, for "white donuts", it seems even worse, with parts of the shape missing that happen to fall onto a different tile:
![3](https://user-images.githubusercontent.com/47149399/51917585-c31e6280-23d7-11e9-827b-62d0ac22d7d9.png)

It would be great if somebody had an idea what went wrong there, even if this issue is already quite old.

**Comment by casualcoder2** [comment]
OK so the disappearing shape parts could be fixed by increasing the padding when drawing the layer, like so:

```js
var pad = new L.Point(this._getMaxRadius(coords.z)*2, this._getMaxRadius(coords.z)*2)
```

This would cause the bounds to grow and more features loaded for a particular map crop. While it may be a bit slower (because it needs to load more features) it makes sure no feature that is relevant gets forgotten.

**Comment by domoritz** [comment]
Hmm, I don't know why the misalignment is happening but it could be a rounding issue in my projections code. I haven't looked at this one years and won't be able to look into it right now. Please send a PR when you find a fix.

**Comment by evgeniiteterin** [comment]
> OK so the disappearing shape parts could be fixed by increasing the padding when drawing the layer, like so:
> 
> ```js
> var pad = new L.Point(this._getMaxRadius(coords.z)*2, this._getMaxRadius(coords.z)*2)
> ```
> 
> This would cause the bounds to grow and more features loaded for a particular map crop. While it may be a bit slower (because it needs to load more features) it makes sure no feature that is relevant gets forgotten.

This trick not work. Did you solve this bug?

**Comment by casualcoder2** [comment]
Nope. My workaround seemed to only work for smaller circles. I stopped looking into it but if you find a solution do let us know!
